x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/index_priority_field.tsx

```diff
@@ -4,23 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { FunctionComponent } from 'react';
+import React, { FunctionComponent, useMemo } from 'react';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { EuiTextColor, EuiDescribedFormGroup } from '@elastic/eui';
-
-import { Phases } from '../../../../../../../common/types';
+import { EuiSpacer, EuiTextColor } from '@elastic/eui';
 
 import { UseField, NumericField } from '../../../../../../shared_imports';
-
-import { LearnMoreLink } from '../..';
+import { LearnMoreLink, DescribedFormRow } from '../..';
+import { i18nTexts } from '../../../i18n_texts';
+import { useEditPolicyContext } from '../../../edit_policy_context';
 
 interface Props {
-  phase: keyof Phases & string;
+  phase: 'hot' | 'warm' | 'cold';
 }
 
-export const SetPriorityInputField: FunctionComponent<Props> = ({ phase }) => {
+export const IndexPriorityField: FunctionComponent<Props> = ({ phase }) => {
+  const { policy, isNewPolicy } = useEditPolicyContext();
+
+  const initialToggleValue = useMemo<boolean>(() => {
+    return (
+      isNewPolicy || // enable index priority for new policies
+      !policy.phases[phase]?.actions || // enable index priority for new phases
+      policy.phases[phase]?.actions?.set_priority != null // enable index priority if it's set
```

Comment:
Hi @jloleysens , here is a change to setting default index priority we discussed. Could you please have a look?

Reply:
Thanks, this makes sense to me! @andreidan @dakrone , this is in line with the current index priority behaviour. I wonder if at this time we can change the past behavior and exclude index priority by default on phases. Currently, it is on for all phases by default for new policies, with defaults for hot being highest, warm second highest and cold lowest.